[CI] EsqlClientYamlIT test {p0=esql/60_enrich/Enrich in fork} failing

**Build Scans:**
- [elasticsearch-periodic #9285 / openjdk21_8.19.0_java-fips-matrix-bwc](https://gradle-enterprise.elastic.co/s/puywiprqi24vc)
- [elasticsearch-periodic #9285 / openjdk21_8.19.0_java-matrix-bwc](https://gradle-enterprise.elastic.co/s/v75zbhyqgwqnc)
- [elasticsearch-periodic #9285 / openjdk23_8.19.0_java-matrix-bwc](https://gradle-enterprise.elastic.co/s/y6hl4e2i2wgrg)
- [elasticsearch-periodic #9285 / 8.19.0_bwc](https://gradle-enterprise.elastic.co/s/bet662h6zatfq)
- [elasticsearch-periodic #9278 / openjdk23_8.19.0_java-matrix-bwc](https://gradle-enterprise.elastic.co/s/fft7wn4mia4g4)
- [elasticsearch-pull-request #80016 / 8.19.0_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/scv2yl7ed3jcq)
- [elasticsearch-pull-request #80045 / 8.19.0_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/yf2dp6yrtnfjm)
- [elasticsearch-pull-request #80042 / 8.19.0_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/k7ktcpcdb5cn6)
- [elasticsearch-pull-request #80016 / 8.19.0_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/idbuu2qgux6lo)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v8.19.0#yamlRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT" -Dtests.method="test {p0=esql/60_enrich/Enrich in fork}" -Dtests.seed=398EB2660E5EFD42 -Dtests.bwc=true -Dtests.locale=nb-NO -Dtests.timezone=Australia/Currie -Druntime.java=21 -Dtests.fips.enabled=true
```

**Applicable branches:**
9.1

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bp0%3Desql%2F60_enrich%2FEnrich%20in%20fork%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [esql/60_enrich:231]: expected [2xx] status code but api [esql.query] returned [400 Bad Request] [{"error":{"root_cause":[{"type":"parsing_exception","reason":"line 1:40: mismatched input 'FORK' expecting {'completion', 'dissect', 'drop', 'enrich', 'eval', 'grok', 'keep', 'limit', 'mv_expand', 'rename', 'sample', 'sort', 'stats', 'where', 'lookup', 'change_point', DEV_INLINESTATS, DEV_LOOKUP, DEV_RERANK, DEV_JOIN_LEFT, DEV_JOIN_RIGHT}","stack_trace":"org.elasticsearch.xpack.esql.parser.ParsingException: line 1:40: mismatched input 'FORK' expecting {'completion', 'dissect', 'drop', 'enrich', 'eval', 'grok', 'keep', 'limit', 'mv_expand', 'rename', 'sample', 'sort', 'stats', 'where', 'lookup', 'change_point', DEV_INLINESTATS, DEV_LOOKUP, DEV_RERANK, DEV_JOIN_LEFT, DEV_JOIN_RIGHT}\n\tat org.elasticsearch.xpack.esql.parser.EsqlParser$1.syntaxError(EsqlParser.java:195)\n\tat org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41)\n
[truncated]
```

**Issue Reasons:**
- [9.1] 4 consecutive failures in test test {p0=esql/60_enrich/Enrich in fork}
- [9.1] 2 consecutive failures in step openjdk23_8.19.0_java-matrix-bwc
- [9.1] 9 failures in test test {p0=esql/60_enrich/Enrich in fork} (50.0% fail rate in 18 executions)
- [9.1] 2 failures in step openjdk23_8.19.0_java-matrix-bwc (100.0% fail rate in 2 executions)
- [9.1] 4 failures in step 8.19.0_part_3_bwc-snapshots (57.1% fail rate in 7 executions)
- [9.1] 2 failures in pipeline elasticsearch-periodic (100.0% fail rate in 2 executions)
- [9.1] 3 failures in pipeline elasticsearch-pull-request (50.0% fail rate in 6 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -576,9 +576,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.action.EsqlActionBreakerIT
   method: testRowStatsProjectGroupByInt
   issue: https://github.com/elastic/elasticsearch/issues/131024
-- class: org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT
-  method: test {p0=esql/60_enrich/Enrich in fork}
-  issue: https://github.com/elastic/elasticsearch/issues/131028
 
 # Examples:
 #

--- a/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_enrich.yml
+++ b/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_enrich.yml
@@ -225,7 +225,7 @@ teardown:
         - method: POST
           path: /_query
           parameters: []
-          capabilities: [ no_brackets_in_unquoted_index_names ]
+          capabilities: [ no_brackets_in_unquoted_index_names, fork_v9 ]
       reason: "Change in the grammar"
 
   - do:

